pkg/algorithm/hived_algorithm_test.go

```diff
@@ -509,6 +592,34 @@ func testSuggestedNodes(t *testing.T, h *HivedAlgorithm) {
 	compareSchedulingResult(t, pod, psr)
 }
 
+func testStatefulPreemption(t *testing.T, configFilePath string) {
+	sConfig := api.NewConfig(api.InitRawConfig(&configFilePath))
+	h := NewHivedAlgorithm(sConfig)
```

Comment:
Also test that A big AG is being preempted by a small AG, the big AG's resource is still hold until it is wholly preempted.